src/basic/store/globalStore.js

```diff
@@ -0,0 +1,39 @@
+import CartList from '../components/CartList';
+
+export const state = (() => {
+  const data = {
+    lastSel: null,
+    bonusPoint: 0,
+    totalAmt: 0,
+    itemCnt: 0,
+    prodList: [
+      { id: 'p1', name: '상품1', price: 10000, volume: 50 },
+      { id: 'p2', name: '상품2', price: 20000, volume: 30 },
+      { id: 'p3', name: '상품3', price: 30000, volume: 20 },
+      { id: 'p4', name: '상품4', price: 15000, volume: 0 },
+      { id: 'p5', name: '상품5', price: 25000, volume: 10 }
+    ],
+    cartList: []
+  };
+
+  const listeners = new Map();
+
+  return {
+    get(key) {
+      return data[key];
+    },
+    set(key, value) {
+      data[key] = value;
+
+      if (listeners.has(key)) {
+        listeners.get(key).forEach((callback) => callback(value));
+      }
+    },
+    subscribe(key, callback) {
+      if (!listeners.has(key)) {
+        listeners.set(key, []);
+      }
+      listeners.get(key).push(callback);
+    }
+  };
+})();
```

Comment:
전역스토어를 구현하신 패턴이 인상적이네요!! 
저는 이런생각까지 못해서, 아마 싱글톤 클래스로 접근했을 것 같은데, 배워갑니다.
리스너를 같이 관리하는 이유도 궁금합니다.(1~주차 과제처럼 전역 addEventLister해줄라고?)
```suggestion
})();
```